### PR TITLE
Navigation: Add 'edit' to the block toolbar

### PR DESCRIPTION
**Pull request by scruffian**
## What?
This adds an "Edit" button the Block Toolbar for the navigation block, which opens the inspector controls.

## Why?
An alternative to https://github.com/WordPress/gutenberg/pull/45772.

## How?
Adds a new plugin to the edit-post package and modifies the edit-site package.

## Testing Instructions
- Add a navigation block in either the site or post editor
- Click on edit in the block toolbar
- Confirm that the block inspector controls open if they aren't already open

## Screenshots or screencast 
Screenshot 2022-11-17 at 13 40 56

## TODO
- [ ] Focus the correct panel inside the inspector controls

### CODE REVIEW

**Comment by getdave** [comment]
I tested and this doesn't seem to be working. Did all changes get pushed?

File: packages/edit-post/src/plugins/navigation-edit-menu-item.js
Comment (getdave) [inline packages/edit-post/src/plugins/navigation-edit-menu-item.js:24]: Is it ok to use string literals here then?

File: packages/edit-post/src/plugins/navigation-edit-menu-item.js
Comment (getdave) [inline packages/edit-post/src/plugins/navigation-edit-menu-item.js:42]: It feels like we should either have:

- normalized the structure of the editors
- create an abstraction to handle this kind of stuff

Not for this PR though...

**Review by getdave — APPROVED** [review]
I tested this and it works 🎉 

Some feedback from a UX perspective which we can iterate on in followups.

1. We should try and focus the `Menu` panel when `Edit` is clicked.
2. Having `Edit` remain active when the sidebar is open feels strange. Maybe it should be disabled? Something about the interaction doesn't feel right to me.

Overall though this now works well and avoids the pitfalls of the other approach.

I have some questions about string literals but I'd need to defer to others on whether the approach is valid or not.


-----

We should get 👍 from @youknowriad before merging this one.

**Comment by getdave** [comment]
@scruffian You should get 👍 from @youknowriad before merging this one.

File: packages/edit-post/src/plugins/navigation-edit-menu-item.js
Comment (youknowriad) [inline packages/edit-post/src/plugins/navigation-edit-menu-item.js:21]: There's no need to test isPostEditor, we're always in post editor here, since this plugin is defined in edit-post

File: packages/edit-site/src/components/block-editor/index.js
Comment (youknowriad) [inline packages/edit-site/src/components/block-editor/index.js:160]: You're removing logic from edit-site here (site editor) without replacing it, is that intended?

Comment (scruffian) [reply packages/edit-site/src/components/block-editor/index.js:160]: Yeah. The button is now created in the edit-post plugin and appears in both places. The old button opened the navigation sidebar. We don't want that at the moment as are experimenting with adding a list view in the navigation  block's inspector controls.

File: packages/edit-post/src/plugins/navigation-edit-menu-item.js
Comment (scruffian) [reply packages/edit-post/src/plugins/navigation-edit-menu-item.js:21]: In my testing, this was also running in the site editor.

File: packages/edit-site/src/components/block-editor/index.js
Comment (youknowriad) [reply packages/edit-site/src/components/block-editor/index.js:160]: Does it mean we're loading edit-post in edit-site? Where are we doing this? That shouldn't be the case.

Comment (scruffian) [reply packages/edit-site/src/components/block-editor/index.js:160]: I was surprised too!

Comment (youknowriad) [reply packages/edit-site/src/components/block-editor/index.js:160]: I think that's a problem that we need to fix and we shouldn't be relying on that to be true here.

Comment (getdave) [reply packages/edit-site/src/components/block-editor/index.js:160]: @scruffian Did we track this down?

**Comment by getdave** [comment]
What are the next steps here?

File: packages/edit-site/src/components/block-editor/index.js
Comment (scruffian) [reply packages/edit-site/src/components/block-editor/index.js:160]: yes it's fixed.

**Comment by scruffian** [comment]
> What are the next steps here?

I think we should probably start afresh. My current thinking is that we should _change_ the list view button in the edit-side component to open the block inspector controls, and add the same thing to edit-post.

**Comment by scruffian** [comment]
Closing in favour of https://github.com/WordPress/gutenberg/pull/46335

**Pull request closed without merging**
